### PR TITLE
percona-server: update livecheck

### PR DESCRIPTION
**Pull request by nandahkrishna**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

Update `percona-server`'s `livecheck`, which was until recently giving an `Unable to get versions` error.

For context, the downloads page now populates versions using the result of a POST request to their versions API:
```
  function fetchVersions(version, cb) {
    jQuery.ajax('/products-api.php', {
      type: 'POST',
      data: { version: version },
      success: function (data, xhr) {
        cb(data);
      },
    });
  }
```
Unless I'm mistaken, we can't do this yet with `livecheck`. Alternatives are to check:
* GitHub tags (https://github.com/percona/percona-server) but I've noticed a lag between the tag and actual release.
* https://docs.percona.com/percona-server/8.0/index.html which links to the release notes for the latest version. This is what I've used. Note that it retrieves older versions too because they're linked from the sidebar.
* https://docs.percona.com/percona-server/8.0/release-notes/release-notes_index.html which lists release notes for all versions.

CC @samford feedback.

I'll create a separate PR for the version bump after this, as I anticipate that I'll need time to debug the issue.

### CODE REVIEW

File: Formula/percona-server.rb [hunk 1]
```diff
@@ -8,8 +8,8 @@ class PerconaServer < Formula
   revision 2
 
   livecheck do
-    url "https://www.percona.com/downloads/Percona-Server-LATEST/"
-    regex(/value=.*?Percona-Server[._-]v?(\d+(?:\.\d+)+-\d+)["' >]/i)
+    url "https://docs.percona.com/percona-server/8.0/index.html"
+    regex(/href=.*?v?(\d+(?:\.\d+)+-\d+)\.html/i)
```
Comment (samford) [inline Formula/percona-server.rb:12]:
```suggestion
    url "https://docs.percona.com/percona-server/latest/index.html"
    regex(/href=.*?v?(\d+(?:[.-]\d+)+)\.html/i)
```

* The `8.0` URL would need to be updated for each major/minor release but substituting `latest` redirects to the correct URL without any versioning issue.
* There have been some tags like `5.1.55`, with no numeric suffix, and `5.6.51-93.0`, with a suffix including a dot. Simply using `[.-]` in the repeating part will handle the current format (e.g., `8.0.33-25`) as well as those previous formats, so there's something to be said for using this approach here.

**Review by samford — COMMENTED** [review]
Thanks, @nandahkrishna! I noticed this a while back but wasn't entirely satisfied with the available options at the time. I meant to revisit the `percona-*` formulae and come up with an interim fix but got busy with other things and forgot 😅

What I saw in researching this before aligns with what you've described and I'm good with using the approach in this PR for the time being.

**Comment by nandahkrishna** [comment]
Ah, good to know that just using `latest` in the URL rather than `8.0` works. Thanks for the suggestions here and in the other PRs!

File: Formula/percona-server.rb [hunk 2]
```diff
@@ -8,8 +8,15 @@ class PerconaServer < Formula
   revision 2
 
   livecheck do
-    url "https://www.percona.com/downloads/Percona-Server-LATEST/"
-    regex(/value=.*?Percona-Server[._-]v?(\d+(?:\.\d+)+-\d+)["' >]/i)
+    url "https://docs.percona.com/percona-server/latest/index.html"
```
Comment (samford) [inline Formula/percona-server.rb:11]:
```suggestion
    url "https://docs.percona.com/percona-server/latest/"
```

I realized right after you committed the suggested changes that we can omit `index.html` from the end (though the trailing forward slash is still required here). It's functionally the same either way but I try to leave off `index.html` when it's not required, just for the sake of simplicity. If it's not too much trouble, we may want to update this and `percona-xtrabackup` one last time before merging.

**Review by samford — APPROVED** [review]

**Pull request merged**